src/xenia/base/system_gnulinux.cc

```diff
@@ -29,7 +33,19 @@ void LaunchWebBrowser(const std::string_view url) {
   system(cmd.c_str());
 }
 
-void LaunchFileExplorer(const std::filesystem::path& path) { assert_always(); }
+void LaunchFileExplorer(const std::filesystem::path& path) {
+  pid_t pid;
+  int n = path.string().length();
+  char cmdxdg[] = "xdg-open";
+  char cmdpath[n + 1];
+  char *argv[] = {cmdxdg, cmdpath, NULL};
+  int status = -1;
+  strcpy(cmdpath, path.c_str());
```

Comment:
Right, So for this specific application using strncpy would be fine? So like:

`strncpy(cmdpath, path.c_str(), path.string().length());`

Reply:
This is equally as insecure, as you're disregarding the size of the _destination_ buffer as much as `strcpy` does. Also, we don't use the non-standard and also dangerous variable-length local arrays. Just create a new `std::string` and gather all the needed parts there, then use `c_str()`.

Reply:
`strncpy` is overall broken by design, by the way, since, if I recall correctly, it doesn't insert the null terminator if overflow has happened. `strlcpy` is the safely truncating function, but it's non-standard. But overall, in Xenia, we don't hesitate to use dynamically allocated temporaries on non-hot paths as long that's more flexible, simple and safe.

Reply:
`std::string.data()` is guaranteed to return a pointer to null-terminated memory since C++11, and since C++17 it returns a mutable string. C++17 is our target, so it should be fine as one of the `argv`.

Reply:
Additionally, the `argv` should be made `const` for clarity (`posix_spawn` takes a `const` for its `argv` argument).